src/main/kotlin/io/easybreezy/hr/application/location/Handler.kt

```diff
@@ -14,17 +14,19 @@ class Handler @Inject constructor(
 ) {
 
     suspend fun handleCreateLocation(command: CreateLocation) = transaction {
-        Location.create(command.name)
+        val location = Location.create(command.name)
+        command.vacationDays?.let { location.updateVacationDays(command.vacationDays) }
     }
 
     suspend fun handleAssignLocation(command: AssignLocation) = transaction {
         val location = locationRepository.getOne(command.locationId)
-        UserLocation.create(
+        val userLocation = UserLocation.create(
             command.startedAt,
             command.endedAt,
             location,
             command.userId
         )
+        command.extraVacationDays?.let { userLocation.addVacationDays(command.extraVacationDays) }
```

Comment:
**User** location means start of work for employee. From this moment he begins to earn vacations days. Spent and not spent days calculates by all user locations and then sum. User location talk that before starting of work user already have had some extra vacation days.

Reply:
Ok, can we hire a member without location? Because our vacations should start calculating after hiring. And what will happen when we change location?

Reply:
We can invite, hire employee and etc. But employee have to have location, it's his start of working. For example, he can start through the week from hiring (he need time to move from other country).
We can change extra vacation days at any moment